### DISCUSSION TITLE
[Suggestions] Changing the resolution of the game in the background while switching from the adventure map to the hero's screen or other similar screens in size

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2449] [Ideas]
When the resolution of the game is, for example, 1024 x 768 and the user opens the hero's screen (or another screen with a smaller original size of 640 x 480), the screen that opens does not occupy the entire screen (as is the case in the original game).

Would it be possible, in the future, to change the resolution of the screen in the background without the user being able to perceive it, to retrieve the same operating mode as the original game for all the resolutions chosen for the adventure map?

**Comment by Branikolog** [comment]
I'm not sure, that stretching the windowed content to fit the screen, while playing on higher resolutions would be eye-friendly to player. Resolution switching doesn't happen at a moment...
I believe, our master developer @ihhub can suggest some kind of scaling to make window look better on high resolutions.

**Reply by LeHerosInconnu** [reply]
Yes, you're right, this should only work for resolutions that have the same screen ratio:
640/480 = 1.333 = 4/3
1024/768 = 1.333 = 4/3
So, it's okay in this case.

So maybe changing resolution in background for the same screen ratio and scaling with black bands for the other cases.
I don't know what would be the most efficient and practical from a programming standpoint and with the best final look.

**Reply by Branikolog** [reply]
Also, I want to mention, that changing resolution in fullscreen mode happens with some sort of blinking. So I'm not sure, that we can do it in background and player wouldn't notice that...

**Reply by LeHerosInconnu** [reply]
Yes, that's also why I asked the question. :)

**Reply by Branikolog** [reply]
So, I think, only our "optimization pro" (@ihhub) Can answer this question. :D

**Reply by sagewitch** [reply]
Yeah, I'm playing on Steam Deck now and unfortunately the size of the window is very small and kind of difficult to see. Hoping there will be an option to adjust the resolution.

**Reply by ihhub** [reply]
@sagewitch ,
Main Menu --> configuration --> graphics --> resolution

Devices with tiny screens need to have as close as possible resolution to the original 640x480 pixels.

**Reply by sagewitch** [reply]
Yeah I already have it set to 1280x800, it's in fullscreen and looks great overall but the resolution of the other screens, like viewing the castle or the hero screen, battle screen, etc. aren't in fullscreen like the original game. So it's more difficult to see on the Steam Deck since the screen on the device is already so small.

**Reply by ihhub** [reply]
@sagewitch , your resolution is already too big for such a tiny screen. Please set 640 x 480 resolution or at max 800 x 600.

**Reply by ihhub** [reply]
We have resolution scaling in our roadmap but as of now the engine does not support it.

**Reply by sagewitch** [reply]
Yeah I already tried making the resolution smaller and it just ended up making the resolution of the screen even smaller. It's kind of an issue with Steam Deck I noticed :( Here's some images of what I mean:

1280x800
https://www.dropbox.com/s/c96rdd9xdiayu96/IMG_0020.HEIC?dl=0
https://www.dropbox.com/s/4oq2qmayskska3g/IMG_0021.HEIC?dl=0

800x600
https://www.dropbox.com/s/uffreqqd0uczsay/IMG_0022.HEIC?dl=0
https://www.dropbox.com/s/2k982qmafqb50w8/IMG_0023.HEIC?dl=0
